pkg/kubelet/pod_workers.go

```diff
@@ -1519,8 +1519,15 @@ func (p *podWorkers) completeWork(podUID types.UID, phaseTransition bool, syncEr
 		// Network is not ready; back off for short period of time and retry as network might be ready soon.
 		p.workQueue.Enqueue(podUID, wait.Jitter(backOffOnTransientErrorPeriod, workerBackOffPeriodJitterFactor))
 	default:
-		// Error occurred during the sync; back off and then retry.
-		p.workQueue.Enqueue(podUID, wait.Jitter(p.backOffPeriod, workerBackOffPeriodJitterFactor))
+		// Error occurred during the sync; back off and then retry. If the error includes a backoff expiration,
+		// wait until then instead of the default to avoid adding extra time to the backoff.
+		var backoffAt time.Time
+		var isBackoffErr bool
+		if backoffAt, isBackoffErr = kubecontainer.MinBackoffExpiration(syncErr); !isBackoffErr {
+			backoffAt = p.clock.Now().Add(p.backOffPeriod)
+		}
+		backoff := backoffAt.Sub(p.clock.Now())
```

Comment:
so perhaps we should test that this is a positive number

Reply:
Yeah, that makes sense. Added the check + test.